src/python/pants/build_graph/address.py

```diff
@@ -151,23 +151,22 @@ def sanitize_path(cls, path: str) -> str:
         if path == "":
             return path
 
-        normpath = os.path.normpath(path)
-        components = normpath.split(os.sep)
-        if components[0] in (".", "..") or normpath != path:
+        components = path.split(os.sep)
+        if any(component in (".", "..", "") for component in components):
```

Comment:
Do you mean to check for `" "`?

Reply:
No, sorry: the description was wrong. This is checking for things like `a//b/c`.